### ISSUE TITLE
Difficulties Reproducing Results

### ISSUE THREAD

**Issue by justinhylee135** [yongliang-wu/DoCo#5]
Dear Yongliang,
Thank you for your previous response. 

I ran the Snoopy checkpoint that you just updated and got the following results:
SnoopyCS: 48.37
MickeyCS: 65.29
SpongebobCS: 70.46

Can you verify that this is similar to the CLIP Scores you received? The scores are definitely much closer to the ones reported in the paper compared to your previous checkpoint, but the MickeyCS is still around 5% lower than what is reported. 

Additionally, I tried using the training arguments you provided with the anchor concept as "mountain" for the target concept Snoopy and I got the following results:
SnoopyCS: 49.69
MickeyCS: 56.56
SpongebobCS: 58.05

The SnoopyCS is similar to the checkpoint you provided but the Mickey and Spongebob CLIP Scores are significantly different. Can you verify that the checkpoint you provided used the following training arguments you commented in the previous [issue](https://github.com/yongliang-wu/DoCo/issues/4)? I'm confused that only 400 steps are used given that your paper says 2000 steps were used.

accelerate launch train.py \
          --pretrained_model_name_or_path=$MODEL_NAME  \
          --output_dir=$OUTPUT_DIR \
          --class_data_dir=./data/samples_mountain/ \
          --class_prompt="mountain" \
          --caption_target "mountain+snoopy" \
          --concept_type object \
          --resolution=512  \
          --train_batch_size=8  \
          --learning_rate=16e-6  \
          --max_train_steps=400 \
          --scale_lr --hflip \
          --parameter_group cross-attn \
          --enable_xformers_memory_efficient_attention \
          --warm_up=200 \
          --with_prior_preservation \
          --dlr 0.0001 \
          --lambda_ 1 \
          --gradient_clip

Finally, could you provide the training arguments for how you handled sequential unlearning? I read your parser argument options and used the flag "--multi_ckpt_path" and passed in your Snoopy checkpoint. I used the same lr, batch, steps, etc. as above to unlearn Mickey and tried both anchor concepts "mouse" and "mountain", however, I got very different results from what was reported in the paper.

Snoopy Checkpoint then Unlearn Mickey with Anchor Mouse:
SnoopyCS: 52.61
MickeyCS: 56.91
Spongebob: 65.554

Snoopy Checkpoint then Unlearn Mickey with Anchor Mountain:
SnoopyCS: 47.52
MickeyCS: 43.22
Spongebob: 48.29

These are very different from {46.53, 47.35, 70.88} CLIP Scores reported in your paper for "Erasing Snoopy and Mickey". Please let me know if I incorrectly followed the procedure for your sequential learning results.

Best,
Justin

**Comment by yongliang-wu** [comment]
Hi, sorry for the delayed response.  

1. For convenience during my reproduction process, I tested it with only 10 templates and obtained results similar to those reported in the paper. In my opinion, The CLIP Score can be quite tricky, so I recommend using other metrics such as BLIP-VQA and incorporating qualitative results for evaluation.  

2. I did train this checkpoint using the parameters I provided in my previous response. Are you still unable to reproduce the results successfully?  

3. The warm-up and training steps may vary depending on different concepts. From my observations, for concepts that are relatively easy to distinguish—such as Snoopy and mountains—you don’t need too many steps to warm up the discriminator since the distinction is quite clear.  

4. You may need to experiment with different hyperparameters to better preserve the concept of Spongebob. Adjusting the training steps could be helpful. DoCo faces similar challenges as GAN in this regard, so I encourage you to try tuning various parameters.
